src/components/v2/Pagination/index.tsx

```diff
@@ -0,0 +1,98 @@
+/** @jsxImportSource @emotion/react */
+import React from 'react';
+import Typography from '@mui/material/Typography';
+import { Button } from '../Button';
+import { Icon, IIconProps } from '../Icon';
+import { useStyles } from './styles';
+
+interface IPaginationProps {
+  className?: string;
+  activePageIndex: number;
+  pagesCount: number;
+  setCurrentPageIndex: (pageIndex: number) => void;
+}
+
+const PAGES_TO_SHOW_COUNT = 4;
+
+export const Pagination = ({
+  className,
+  activePageIndex,
+  pagesCount,
+  setCurrentPageIndex,
+}: IPaginationProps) => {
+  const classes = useStyles();
+
+  /* creating pages array */
+  const pagesArray = Array.from({ length: pagesCount }, (_, i) => i + 1);
+
+  const halfOfPagesCount = Math.ceil(PAGES_TO_SHOW_COUNT / 2);
+  const lastPageIndex = pagesCount - 1;
+  const isActivePageInEnd = activePageIndex > lastPageIndex - halfOfPagesCount;
+  const isActivePageInStart = activePageIndex < halfOfPagesCount;
+
+  const minPageIndexToShow = isActivePageInEnd
+    ? lastPageIndex - PAGES_TO_SHOW_COUNT
+    : activePageIndex - halfOfPagesCount;
+
+  const maxPageIndexToShow = isActivePageInStart
+    ? PAGES_TO_SHOW_COUNT
+    : activePageIndex + halfOfPagesCount;
+
+  if (pagesCount <= 1) {
+    return null;
+  }
+
+  return (
+    <div css={[classes.root, className]}>
+      {pagesArray.map((page, index) => {
+        const iconProps: IIconProps = { size: '20px', name: 'arrowRight', color: 'inherit' };
```

Comment:
We've established we want to avoid using inline styles: you can set the size (width + height) from the styles file.

Reply:
there are props, but not inline styles

Reply:
and if we want to avoid the `size` prop, we should remove it from `IIconProps`

Reply:
We can yeah